### PR TITLE
Emoji reacts open group poll & update conflict handling

### PR DESCRIPTION
**Pull request by RyanRory**

### CODE REVIEW

File: Session/Notifications/AppNotifications.swift
Comment (mpretty-cyro) [inline Session/Notifications/AppNotifications.swift:392]: Unrelated changes?

File: SessionMessagingKit/Messages/Message.swift
Comment (mpretty-cyro) [inline SessionMessagingKit/Messages/Message.swift:388]: This can be cleaned up a bit to:
```
let shouldAddSelfReaction: Bool = (
    rawReaction.you ||
    reactors.contains(userPublicKey) ||
    associatedPendingChanges
        .contains { pendingChange in
            if case .reaction(_, let emoji, let action) = pendingChange.metadata {
                return (
                    emoji == decodedEmoji &&
                    action == .react
                )
            }
            
            return false
        }
)
```

File: SessionMessagingKit/Open Groups/OpenGroupManager.swift
Comment (mpretty-cyro) [inline SessionMessagingKit/Open Groups/OpenGroupManager.swift:545]: Do we want to remove entries with no `seqNo` here? (that would include any in-flight changes)

File: SessionMessagingKit/Sending & Receiving/MessageReceiver.swift
Comment (mpretty-cyro) [inline SessionMessagingKit/Sending & Receiving/MessageReceiver.swift:258]: Ah good catch lol

**Review by mpretty-cyro — CHANGES_REQUESTED** [review]

File: Session/Notifications/AppNotifications.swift
Comment (RyanRory) [reply Session/Notifications/AppNotifications.swift:392]: Can't compile without this change after merging dev branch.

File: SessionMessagingKit/Messages/Message.swift
Comment (RyanRory) [reply SessionMessagingKit/Messages/Message.swift:388]: Will clean up this.

File: SessionMessagingKit/Open Groups/OpenGroupManager.swift
Comment (RyanRory) [reply SessionMessagingKit/Open Groups/OpenGroupManager.swift:545]: Hmm... I'm keeping those entries with no `seqNo` or `seqNo` greater than current polling's `seqNo`

File: Session/Notifications/AppNotifications.swift
Comment (mpretty-cyro) [reply Session/Notifications/AppNotifications.swift:392]: Yep just noticed this - sorry about that

File: SessionMessagingKit/Open Groups/OpenGroupManager.swift
Comment (mpretty-cyro) [reply SessionMessagingKit/Open Groups/OpenGroupManager.swift:545]: Oh sorry I misread this - you are right this code is correct 😅

File: SessionMessagingKit/Messages/Message.swift
Comment (RyanRory) [reply SessionMessagingKit/Messages/Message.swift:388]: Hmm... Take a closer look and the logic can't be changed like this. Because the logic needs to be when there is no pending changes, the source of truth will be `rawReaction.you || reactors.contains(userPublicKey)`, if there are pending changes, the the source of truth will be the last pending changes and what the server returns will be ignored.

Comment (mpretty-cyro) [reply SessionMessagingKit/Messages/Message.swift:388]: Ah sorry I missed that we were also forcing the result to `false` for the `remove` case 😞 

The main thing I'm looking to do is remove the mutability of the `shouldAddSelfReaction` variable so if a future dev modifies the code further in that function they need to explicitly make the variable mutable again (the current intent of this code is that after it is set the `shouldAddSelfReaction` shouldn't be getting updated) - it's not as succinct but the below should handle both cases

This also includes explicit sorting to deal with situations where the user adds/removes reactions and ends up with multiple pending changes for the same reaction (it'll be a bit weird while the add/remove API call hasn't returned but should handle the case were we are waiting on a response from the poller a little better)
```
let pendingChangeSelfReaction: Bool? = {
    // Find the newest 'PendingChange' entry with a matching emoji, if one exists, and
    // set the "self reaction" value based on it's action
    let maybePendingChange: PendingChange? = associatedPendingChanges
        .sorted(by: { lhs, rhs -> Bool in (lhs.seqNo ?? Int64.max) > (rhs.seqNo ?? Int64.max) })
        .first { pendingChange in
            if case .reaction(_, let emoji, _) = pendingChange.metadata {
                return emoji == decodedEmoji
            }
            
            return false
        }
    
    // If there is no pending change for this reaction then return nil
    guard
        let pendingChange: PendingChange = maybePendingChange,
        case .reaction(_, _, let action) = pendingChange.metadata
    else { return nil }

    // Otherwise add/remove accordingly
    return (action == .react)
}()
let shouldAddSelfReaction: Bool = (
    pendingChangeSelfReaction ??
    (rawReaction.you || reactors.contains(userPublicKey))
)
```

Comment (mpretty-cyro) [reply SessionMessagingKit/Messages/Message.swift:388]: Just for reference - when I add code snippets I don't generally run it to confirm it's working correctly, it's mostly just recommendations so always best to double check it makes sense 😜

**Review by mpretty-cyro — APPROVED** [review]

**Pull request merged**
